### PR TITLE
HentaiRead | Fixed No Results Found, Added Filters

### LINKED ISSUE #4998: Hentai Read "No Results Found" and "HTTP 404" but Accessible to Webview
### Source information

HentaiRead 1.4.42

### Source language

English

### Steps to reproduce

1. Go to Browse/ Got to Library pick any HentaiRead Manga to read
2. Access the Extension/Press the Chapter
3. "No Results Found", "Failed to Find Page List. Non-English entries are not supported" and "HTTP 404"

### Expected behavior

This should happen got access to the Doujins like it did Yesterday

### Actual behavior

This happened instead "No Results Found", "Failed to Find Page List. Non-English entries are not supported" and "HTTP 404"

### Mihon/Tachiyomi version

Mihon 0.16.5 and Tachiyomi Stable 0.15.3

### Android version

Android 10

### Other details

att PsPL7sZELGO0ZieJ8q6IxwFg7BORjtTtb4i5Uf4Tjx4
att 6fxIN-ribdF8WHR3-GA8YUe2JJ15WdwhUNiH19HfJAo
att qONUAl81S_lK-BWjXeQ5A02iSj2__ucXvQJ3iCtHlpY1

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by KenjieDec**
Closes #4998

Checklist:
- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/en/hentairead/src/eu/kanade/tachiyomi/extension/en/hentairead/Hentairead.kt [hunk 1]
```diff
@@ -50,31 +91,108 @@ class Hentairead : Madara("HentaiRead", "https://hentairead.com", "en", dateForm
...
-    override val mangaDetailsSelectorTag = "div.post-meta.post-tax-wp-manga-tag > span.post-tags > a > span.tag-name"
-
-    override val pageListParseSelector = "li.chapter-image-item > a > div.image-wrapper"
+    override fun popularMangaRequest(page: Int): Request = GET("$baseUrl/$mangaSubString/${searchPage(page)}?sortby=views", headers)
+    override fun latestUpdatesRequest(page: Int): Request = GET("$baseUrl/$mangaSubString/${searchPage(page)}?sortby=new", headers)
+    override fun popularMangaSelector() = ".manga-item"
+    override val popularMangaUrlSelector = ".manga-item__bottom a"
+
+    private fun getTagId(tag: String, type: String): Int? {
+        val ajax = "https://hentairead.com/wp-admin/admin-ajax.php?action=search_manga_terms&search=$tag&taxonomy=$type".replace("artist", "manga_artist")
```
Comment (AwkwardPeak7) [inline src/en/hentairead/src/eu/kanade/tachiyomi/extension/en/hentairead/Hentairead.kt]: baseurl

File: src/en/hentairead/src/eu/kanade/tachiyomi/extension/en/hentairead/Hentairead.kt [hunk 2]
```diff
@@ -50,31 +91,108 @@ class Hentairead : Madara("HentaiRead", "https://hentairead.com", "en", dateForm
...
+                            addQueryParameter("release-type", type)
+                            addQueryParameter("release", it.state.filter(Char::isDigit))
+                        }
+                    }
+
+                    is TextFilter -> {
+                        if (it.state.isNotEmpty()) {
+                            it.state.split(",").filter(String::isNotBlank).map { tag ->
+                                val trimmed = tag.trim()
+                                val id = getTagId(trimmed.removePrefix("-"), it.type)?.toString()
```
Comment (AwkwardPeak7) [inline src/en/hentairead/src/eu/kanade/tachiyomi/extension/en/hentairead/Hentairead.kt:145]: would this make n calls for n tags?

Comment (KenjieDec) [reply src/en/hentairead/src/eu/kanade/tachiyomi/extension/en/hentairead/Hentairead.kt:145]: > would this make n calls for n tags?

Yes, I have yet to find a way to get all of the tags' ids at once.

**Review by AwkwardPeak7 — APPROVED** [review]

**Comment by shuukenji92** [comment]
Thank you both for your hardwork. I am truly grateful :)

**Pull request merged**